Source/LinkedList.c

```diff
@@ -2,5 +2,48 @@
  * @file
  * @brief Implementation of singly-linked list.
  */
-
+#include <stdio.h>
+#include <stdlib.h>
 #include "LinkedList.h"
+
+
+void LinkedList_Init(LinkedList_t *instance)
+{
+   instance->head = NULL;
+}
+
+int LinkedList_Count(LinkedList_t *instance)
+{
+   LinkedListNode_t *current = instance->head;
+   int count = 0;
+   while(current != NULL)
+   {
+      count++;
+      current = current->next;
+   }
+   return count;
+}
+
+void LinkedList_PushFront(LinkedList_t *instance, LinkedListNode_t *node)
+{
+   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
+   LinkedListNode_t *tempNode = instance->head;
+   instance->head = node;
+   node->next = tempNode;
+}
+
+void LinkedList_PushBack(LinkedList_t *instance, LinkedListNode_t *node)
+{
+   node = (LinkedListNode_t *)malloc(sizeof(LinkedListNode_t));
```

Comment:
A linked list node is passed to this function already. You can use it instead of `malloc`ing a new node. We take this strategy here because it 1) allows us to run something like this in a heapless environment and 2) allows the client to allocate a "sub-class" of `LinkedListNode_t` so that the client can store arbitrary date in the linked list.

If you do choose to use `malloc`, be sure to create another way for the client to store arbitrary date in the linked list and remove the `node` parameter from the parameter list.

Reply:
So do I allocate space in the tests for the nodes? I'm confused on how its possible to add it to the list if there is no space assigned to it in memory. If I were to remove the malloc, then the program does not run.

Reply:
are you referring to something like this? https://stackoverflow.com/questions/3856317/linked-lists-in-c-without-malloc